src/openrct2/object/SceneryGroupObject.cpp

```diff
@@ -118,8 +118,7 @@ void SceneryGroupObject::UpdateEntryIndexes()
         auto sceneryType = GetSceneryType(ori->Type);
         if (sceneryType.has_value())
         {
-            _legacyType.scenery_entries[_legacyType.entry_count] = { sceneryType.value(), entryIndex };
-            _legacyType.entry_count++;
+            _legacyType.SceneryEntries.emplace_back(sceneryType.value(), entryIndex);
```

Comment:
```suggestion
            _legacyType.SceneryEntries.push_back({ sceneryType.value(), entryIndex });
```

Reply:
`ScenerySelection` is a `struct` with no user defined ctors, so emplace back did not work and thus I reverted it.